src/txmempool.cpp

```diff
@@ -212,6 +244,7 @@ void CTxMemPool::removeForBlock(const std::vector<CTransaction>& vtx, unsigned i
     BOOST_FOREACH(const CTransaction& tx, vtx)
     {
         std::list<CTransaction> dummy;
+        updateDependentPriorities(tx, nBlockHeight, true);
```

Comment:
If a block contains transactions that depend on other transactions in the same block, won't this cause recalcPriority to be called multiple times for them?